OSSpecific/src/artofillusion/osspecific/MacOSPlugin.java

```diff
@@ -13,96 +13,86 @@ Changes copyright (C) 2017 by Maksim Khramov
+                System.out.println("Unable to instantiate JDK9 adapter.");
+            }
+        }
 
-        ex.printStackTrace();
-      }
-      usingAppMenu = true;
     }
-    else if (message == SCENE_WINDOW_CREATED)
+
+    @Override
+    protected void onSceneSaved(File file, LayoutWindow view)
     {
-      final LayoutWindow win = (LayoutWindow) args[0];
-      win.addEventLink(SceneChangedEvent.class, new Object() {
-        void processEvent()
+        if(isMac)
         {
-          updateWindowProperties(win);
+            updateWindowProperties(view);
+            view.getComponent().getRootPane().putClientProperty("Window.documentModified", false);
         }
-      });
-      updateWindowProperties(win);
-      if (!usingAppMenu)
-        return;
-
-      // Remove the Quit and Preferences menu items, since we are using the ones in the application
-      // menu instead.
-
-      removeMenuItem(win, Translate.text("menu.file"), Translate.text("menu.quit"));
-      removeMenuItem(win, Translate.text("menu.edit"), Translate.text("menu.preferences"));
     }
-    else if (message == SCENE_SAVED)
-    {
-      LayoutWindow win = (LayoutWindow) args[1];
-      updateWindowProperties(win);
-      win.getComponent().getRootPane().putClientProperty("Window.documentModified", false);
+
+    @Override
+    protected void onSceneWindowCreated(final LayoutWindow view) {
+        if(isMac)
+        {
+            view.addEventLink(SceneChangedEvent.class, new Object() {
+                void processEvent() {
+                    updateWindowProperties(view);
+                }
+            });
+            updateWindowProperties(view);
+            view.getComponent().getRootPane().putClientProperty("Window.documentModified", false);
+            // Remove the Quit and Preferences menu items, since we are using the ones in the application
+            // menu instead.
+
+            removeMenuItem(view, Translate.text("menu.file"), Translate.text("menu.quit"));
+            removeMenuItem(view, Translate.text("menu.edit"), Translate.text("menu.preferences"));
+        }
     }
-  }
 
   /** Update the Mac OS X specific client properties. */
 
-  private void updateWindowProperties(LayoutWindow win)
+  private static void updateWindowProperties(LayoutWindow view)
   {
-    win.getComponent().getRootPane().putClientProperty("Window.documentModified", win.isModified());
-    Scene scene = win.getScene();
-    if (scene.getName() != null)
-    {
-      File file = new File(scene.getDirectory(), scene.getName());
-      win.getComponent().getRootPane().putClientProperty("Window.documentFile", file);
-    }
+    view.getComponent().getRootPane().putClientProperty("Window.documentModified", view.isModified());
+    Scene scene = view.getScene();
+    if(null == scene.getName()) return;
```

Comment:
Please don't use void early-return like this.
`if(guard) return;` really messes with how source is read.

When I see an if, I'm expecting

```
if(okaytowork)
{
  Do stuff;
}
```